pro/src/screens/Desk/Desk.tsx

```diff
@@ -0,0 +1,192 @@
+import cx from 'classnames'
+import React, { useState, useRef } from 'react'
+
+import TextInput from 'components/layout/inputs/TextInput/TextInput'
+import Titles from 'components/layout/Titles/Titles'
+import { Banner, Button } from 'ui-kit'
+
+import { BookingDetails } from './BookingDetails'
+import { ButtonInvalidateToken } from './ButtonInvalidateToken'
+import styles from './Desk.module.scss'
+import {
+  MESSAGE_VARIANT,
+  IErrorMessage,
+  IBooking,
+  IDeskGetBookingResponse,
+  IDeskProps,
+  IDeskSubmitResponse,
+} from './types'
+import { validateToken } from './validation'
+
+const Desk = ({
+  getBooking,
+  submitInvalidate,
+  submitValidate,
+}: IDeskProps): JSX.Element => {
+  const [token, setToken] = useState('')
+  const [isTokenValidated, setIsTokenValidated] = useState(false)
+  const [booking, setBooking] = useState<IBooking | null>(null)
+  const [message, setMessage] = useState<IErrorMessage>({
+    message: 'Saisissez une contremarque',
+    variant: MESSAGE_VARIANT.DEFAULT,
+  })
+  const [disableSubmitValidate, setDisableSubmitValidate] = useState(true)
+  const tokenInputRef = useRef<HTMLInputElement | null>(null)
+
+  const resetMessage = () => {
+    setMessage({
+      message: 'Saisissez une contremarque',
+      variant: MESSAGE_VARIANT.DEFAULT,
+    })
+  }
+
+  const resetTokenField = () => {
+    if (tokenInputRef.current !== null) {
+      tokenInputRef.current.value = ''
+      tokenInputRef.current.focus()
+    }
+    setDisableSubmitValidate(true)
+  }
+
+  const handleOnChangeToken = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value.toUpperCase()
+    // QRCODE return a prefix that we want to ignore.
+    const token = inputValue.split(':').reverse()[0]
+
+    setToken(token)
+    setIsTokenValidated(false)
+    setMessage({
+      message: 'Vérification...',
+      variant: MESSAGE_VARIANT.DEFAULT,
+    })
```

Comment:
Ici on se retrouve à changer 3 states différents après la même action --> ces states sont liés
Dans ces cas là, je trouve que le hook de react `useReducer` s'adapte bien https://reactjs.org/docs/hooks-reference.html#usereducer

Reply:
Je ne vois pas l'interet, ils peuvent etre edité ensemble mais pas forcement.

Reply:
```
useReducer is usually preferable to useState when you have complex state logic that involves multiple sub-values or when the next state depends on the previous one. useReducer also lets you optimize performance for components that trigger deep updates because [you can pass dispatch down instead of callbacks](https://reactjs.org/docs/hooks-faq.html#how-to-avoid-passing-callbacks-down).
```
Ca ne semble pas trop adapté au cas present.